src/VirtualClient/VirtualClient.Actions/SuperBenchmark/SuperBenchmarkExecutor.cs

```diff
@@ -151,18 +165,47 @@ protected override async Task ExecuteAsync(EventContext telemetryContext, Cancel
         /// </summary>
         protected override async Task InitializeAsync(EventContext telemetryContext, CancellationToken cancellationToken)
         {
-            SuperBenchmarkState state = await this.stateManager.GetStateAsync<SuperBenchmarkState>($"{nameof(SuperBenchmarkState)}", cancellationToken)
+            var repositoryLinkUri = new Uri(this.RepositoryLink);
+            this.repositoryName = Path.GetFileName(repositoryLinkUri.AbsolutePath);
+            
+            // download config file if a link is provided
+            if (this.ConfigurationFile.StartsWith("http"))
+            {                
+                var configFileUri = new Uri(this.ConfigurationFile);
+                string configFileName = Path.GetFileName(configFileUri.AbsolutePath);
+                string configFullPath = this.PlatformSpecifics.Combine(this.SuperBenchmarkDirectory, configFileName);
+
+                using (var client = new HttpClient())
```

Comment:
Let's not add un-unit testable code here.

Reply:
Are cmd line based methods preferred? like wget or curl?